packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Writers/CodeWriter.cs

```diff
@@ -267,6 +267,40 @@ public void WriteMethod(CSharpMethod method)
             CodeModelPlugin.Instance.CodeWriterExtensionMethods.WriteMethod(this, method);
         }
 
+        public CodeWriter WriteMethodDocumentation(MethodSignatureBase methodBase)
+        {
+            if (methodBase.IsRawSummaryText)
+            {
+                return WriteRawXmlDocumentation(methodBase.Description);
+            }
+
+            if (methodBase.NonDocumentComment is { } comment)
+            {
+                WriteLine($"// {comment}");
+            }
+
+            if (methodBase.SummaryText is { } summaryText)
+            {
+                WriteXmlDocumentationSummary(summaryText);
+            }
+
+            return WriteMethodDocumentationSignature(methodBase);
+        }
+
+        public CodeWriter WriteMethodDocumentationSignature(MethodSignatureBase methodBase)
+        {
+            WriteXmlDocumentationParameters(methodBase.Modifiers.HasFlag(MethodSignatureModifiers.Public) ? methodBase.Parameters : methodBase.Parameters.Where(p => p.Description is not null));
+
+            WriteXmlDocumentationRequiredParametersException(methodBase.Parameters);
+            WriteXmlDocumentationNonEmptyParametersException(methodBase.Parameters);
+            if (methodBase is MethodSignature { ReturnDescription: { } } method)
+            {
```

Comment:
I think we do not have to rush this because this is tracking in this issue for writing documents: https://github.com/Azure/autorest.csharp/issues/3922

Reply:
I just moved this from autorest.csharp. We can update the documents later as a part of this issue.